modules/gitlab_integration/spec/lib/open_project/gitlab_integration/notification_handler/push_hook_spec.rb

```diff
@@ -0,0 +1,151 @@
+          "url" => "git@c7e7cd2d54c3:openprojecttest/test.git",
+          "ssh_url" => "git@c7e7cd2d54c3:openprojecttest/test.git",
+          "http_url" => "http://c7e7cd2d54c3/openprojecttest/test.git"
+        },
+      "commits" => [
+        {
+          "id" => "a265d6b7bcf836b77ed9e32f824b231585c6a355",
+          "message" => commit_message,
+          "title" => commit_title,
+          "timestamp" => "2024-07-22T11:18:29+02:00",
+          "url" => "http://c7e7cd2d54c3/openprojecttest/test/-/commit/a265d6b7bcf836b77ed9e32f824b231585c6a355",
+          "author" => { "name" => "Some committer", "email" => "some_committer@example.com" },
+          "added" => [],
+          "modified" => ["CHANGELOG"],
+          "removed" => []
+        }
+      ],
+      "total_commits_count" => 1,
+      "push_options" => {},
+      "repository" =>
+        {
+          "name" => "Test",
+          "url" => "git@c7e7cd2d54c3:openprojecttest/test.git",
+          "description" => nil,
+          "homepage" => "http://c7e7cd2d54c3/openprojecttest/test",
+          "git_http_url" => "http://c7e7cd2d54c3/openprojecttest/test.git",
+          "git_ssh_url" => "git@c7e7cd2d54c3:openprojecttest/test.git",
+          "visibility_level" => 10
+        },
+      "open_project_user_id" => gitlab_system_user.id,
+      "gitlab_event" => "push_hook"
+    }
+  end
+
+  before do
+    allow(handler_instance).to receive(:comment_on_referenced_work_packages).and_return(nil)
+  end
+
+  context "with a regular push" do
+    let(:comment) do
+      "**Pushed in refs/heads/main:** [Administrator]" \
+        "(https://www.gravatar.com/avatar/65a222b844ced567fe0ed2594c0b4abdf62efa1322a385c919c41e7bbc16d4fc?s=80&d=identicon) " \
+        "pushed [a265d6b7](http://c7e7cd2d54c3/openprojecttest/test/-/commit/a265d6b7bcf836b77ed9e32f824b231585c6a355) " \
+        "to [Test](http://c7e7cd2d54c3/openprojecttest/test) at 2024-07-22T11:18:29+02:00:" \
+        "\nMentioning OP##{work_package.id}\n\nSome commit message\n\n"
+    end
+
+    it "adds a comment to the work packages" do
+      process
+      expect(handler_instance).to have_received(:comment_on_referenced_work_packages).with(
+        [work_package],
+        gitlab_system_user,
+        comment
+      )
+    end
+
+    context "when no commit message is given in the payload" do
+      before do
+        payload["commits"][0]["message"] = nil
+      end
+
+      let(:comment) do
+        "**Pushed in refs/heads/main:** [Administrator]" \
+          "(https://www.gravatar.com/avatar/65a222b844ced567fe0ed2594c0b4abdf62efa1322a385c919c41e7bbc16d4fc?s=80&d=identicon) " \
+          "pushed [a265d6b7](http://c7e7cd2d54c3/openprojecttest/test/-/commit/a265d6b7bcf836b77ed9e32f824b231585c6a355) " \
+          "to [Test](http://c7e7cd2d54c3/openprojecttest/test) at 2024-07-22T11:18:29+02:00:\nMentioning OP##{work_package.id}\n"
+      end
+
+      it "does not raise (Bugfix)" do
```

Comment:
🟡 While the test mentions not raising, there's no assertion to it. I get why but I'd prefer for it either have one assertion like `expect { process }.to_not raise_error(ExceptionOfInterest)` and the message expectation or the description changed to something along the lines of `handles commits without messages`

Reply:
Good point. When writing my thinking was that the test would raise and therefore "fail" if there still was an error, but I also agree that making things explicit is often the better choice.
So I added your suggested expectation, but since you already approved I assume you don't want to review again, right? :slightly_smiling_face:

Reply:
That was my only point and wasn't something that I feel should even block the merge, so feel free to :shipit: =)